File: Accupay.Data/Entities/Bonus.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;
using System.Linq;

namespace AccuPay.Data.Entities
{
    [Table("employeebonus")]
    public class Bonus
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? RowID { get; set; }

        public int? OrganizationID { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public DateTime Created { get; set; }

        public int? CreatedBy { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime? LastUpd { get; set; }

        public int? LastUpdBy { get; set; }

        public int? EmployeeID { get; set; }

        public int? ProductID { get; set; }

        [ForeignKey("ProductID")]
        public virtual Product Product { get; set; }

        [Column("EffectiveStartDate")]
        public DateTime EffectiveStartDate { get; set; }

        [Column("EffectiveEndDate")]
        public DateTime EffectiveEndDate { get; set; }

        [Column("AllowanceFrequency")]
        public string AllowanceFrequency { get; set; }

        [Column("TaxableFlag")]
        public string TaxableFlag { get; set; }

        [Column("BonusAmount")]
        public decimal? BonusAmount { get; set; }

        [Column("RemainingBalance")]
        public decimal? RemainingBalance { get; set; }

        public string BonusType
        {
            get
            {
                return Product?.Name;
            }
        }
    }
}

File: AccuPay/Domain/Paystubs/Bonus.vb
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Global.AccuPay.Entity

    <Table("employeebonus")>
    Public Class Bonus

        <Key>
        Public Property RowID As Integer

        Public Property OrganizationID As Integer

        <DatabaseGenerated(DatabaseGeneratedOption.Computed)>
        Public Property Created As DateTime

        Public Property CreatedBy As Integer?

        <DatabaseGenerated(DatabaseGeneratedOption.Computed)>
        Public Property LastUpd As DateTime?

        Public Property LastUpdBy As Integer?
        Public Property EmployeeID As Integer?
        Public Property ProductID As Integer?
        Public Property EffectiveStartDate As Date?
        Public Property AllowanceFrequency As String
        Public Property EffectiveEndDate As Date?
        Public Property TaxableFlag As String
        Public Property BonusAmount As Decimal?
        Public Property RemainingBalance As Decimal?

    End Class

End Namespace